src/core/events-grant-authorization.ts

```diff
@@ -0,0 +1,44 @@
+import type { EventsQueryMessage } from '../types/events-types.js';
+import type { MessageStore } from '../types/message-store.js';
+import type { PermissionGrant } from '../protocols/permission-grant.js';
+
+import { GrantAuthorization } from './grant-authorization.js';
+import { DwnError, DwnErrorCode } from './dwn-error.js';
+
+export class EventsGrantAuthorization {
+  /**
+   * Authorizes the given EventsQuery in the scope of the DID given.
+   */
+  public static async authorizeQuery(input: {
+    eventsQueryMessage: EventsQueryMessage,
+    expectedGrantor: string,
+    expectedGrantee: string,
+    permissionGrant: PermissionGrant,
+    messageStore: MessageStore,
+  }): Promise<void> {
+    const {
+      eventsQueryMessage, expectedGrantor, expectedGrantee, permissionGrant, messageStore
+    } = input;
+
+    await GrantAuthorization.performBaseValidation({
+      incomingMessage: eventsQueryMessage,
+      expectedGrantor,
+      expectedGrantee,
+      permissionGrant,
+      messageStore
+    });
+
+    // if the grant is scoped to a specific protocol, ensure that all of the query filters must include that protocol
+    if ('protocol' in permissionGrant.scope && permissionGrant.scope.protocol !== undefined) {
```

Comment:
Optional: Is it possible/worth it to use `hasProtocolScope()`, just a thought, ignore if painful or not worth it.

Reply:
Yup! Totally missed the opportunity for that, no circular deps so it's good!